pkg/controller/master/storagenetwork/storage_network.go

```diff
@@ -298,34 +300,55 @@ func (h *Handler) createNad(setting *harvesterv1.Setting) (string, error) {
 	nad.Annotations = map[string]string{
 		StorageNetworkAnnotation: "true",
 	}
+	nad.Labels = map[string]string{
+		HashStorageNetworkLabel: h.sha1(setting.Value),
+	}
 	nad.Spec.Config = string(nadConfig)
 
 	// create nad
 	var nadResult *nadv1.NetworkAttachmentDefinition
 	if nadResult, err = h.networkAttachmentDefinitions.Create(&nad); err != nil {
-		return "", fmt.Errorf("create net-attach-def failed %v", err)
+		return nil, fmt.Errorf("create net-attach-def failed %v", err)
+	}
+
+	return nadResult, nil
+}
+
+func (h *Handler) findOrCreateNad(setting *harvesterv1.Setting) (*nadv1.NetworkAttachmentDefinition, error) {
+	sets := labels.Set{
+		HashStorageNetworkLabel: h.sha1(setting.Value),
+	}
+
+	nads, err := h.networkAttachmentDefinitionsCache.List(StorageNetworkNetAttachDefNamespace, sets.AsSelector())
+	if err != nil {
+		return nil, err
+	}
+
+	if len(nads) > 0 {
```

Comment:
Do we want to handle the case of more than one NAD? Or print a warning message because those excessive NADs should be safe enough to be purged manually.

Reply:
Updated per suggestion, if there is more than one match NAD, the controller will leave a warning message and choose the first one. Thanks.